src/components/App.js

```diff
@@ -1,15 +1,118 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { BrowserRouter, Route } from 'react-router-dom';
+import { getWeatherInformation } from '../utils/weather';
+import getLocation from '../utils/location';
+
+import AirQuality from './AirQuality';
+import Forecast from './Forecast';
+import Nav from './Nav';
+import UnitToggle from './UnitToggle';
 import Weather from './Weather';
+/* eslint-disable react/no-unused-state */
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      isCelcius: false,
+      location: false,
+      latitude: null,
+      longitude: null,
+      weatherData: null,
+      forecastData: null,
+      forecastResponse: false,
+      weatherResponse: false,
+      temperature: null,
+    };
+
+    this.handleUnitChange = this.handleUnitChange.bind(this);
+  }
+
+  componentDidMount() {
+    if ('geolocation' in navigator) {
+      getLocation().then(response => {
+        this.setState({
+          location: true,
+          latitude: response.coords.latitude,
+          longitude: response.coords.longitude,
+        });
+        this.getCurrentWeather();
+        this.getCurrentForecast();
+      });
+    }
+  }
+
+  getCurrentWeather() {
+    const { latitude, longitude, location } = this.state;
+    const url = `http://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=${
```

Comment:
I think you can maybe create a function `getServiceUrl(service)` that returns this url and where `service` is either `weather` or `forecast/daily`

Reply:
done